edit-post/components/browser-url/index.js

```diff
@@ -26,8 +42,13 @@ export class BrowserURL extends Component {
 	}
 
 	componentDidUpdate( prevProps ) {
-		const { postId, postStatus } = this.props;
+		const { postId, postStatus, postType } = this.props;
 		const { historyId } = this.state;
+
+		if ( postStatus === 'trashed' ) {
```

Comment:
The logic flow for this lifecycle function is a bit confusing, since it handles 2 cases (with 3 conditions including a `return`) for which one but not the other is delegated to a separate function (`setBrowserURL`).

Should we consolidate to two conditions (one per case), and use separate functions consistently?

```js
componentDidUpdate( prevProps ) {
	const { postId, postStatus, postType } = this.props;

	if ( postStatus === 'trashed' ) {
		this.setTrashURL( postId, postType );
	}

	const { historyId } = this.state;
	if ( ( postId !== prevProps.postId || postId !== historyId ) && postStatus !== 'auto-draft' ) {
		this.setBrowserURL( postId );
	}
}
```

Or maybe some other pattern for handling multiple types of lifecycle changes, e.g. `componentDidUpdate` assigned by [`_.over`](https://lodash.com/docs/4.17.10#over) or [`_.cond`](https://lodash.com/docs/4.17.10#cond).

Similar to:

https://github.com/WordPress/gutenberg/blob/5d7bc408774d79219fd917f0e53d937633720eaf/editor/components/navigable-toolbar/index.js#L33-L35

Reply:
I'm using the less smart option :P. I think it's a little bit easier to understand.